src/audio/rtnr/rtnr.c

```diff
@@ -287,7 +293,14 @@ static struct comp_dev *rtnr_new(const struct comp_driver *drv,
 	/* Component defaults */
 	cd->source_channel = 0;
 
-    /* Check config */
+	/* TODO(Realtek): it doesn't make sense to fetch config during rtnr_new() because
+	 *                new() is always the first function called for the component. The
+	 *                fetched config here is the default config specified in topology.
+	 *                I suspect that the config blob given in UCM is the one RTNR
+	 *                requires, instead of the default one.
+	 *                Please move to rtnr_prepare().
+	 */
+	/* Check config */
```

Comment:
This is because RTNR requires a default sample rate while creating instance.

Reply:
If so I would suggest to defer RTKMA_API_Context_Create() to rtnr_params(), where we could get the sample rate from source and sink buffer without config blob, unless the required sample rate is neither of them. Even so, I am still of the opinion against fetching config in rtnr_new(). Could we use a constant value for the default sample rate?
@cujomalainey WDYT?

Reply:
Could we simply instantiate on the first call to params and then only recreate as needed if the next params call has a different rate?